Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -438,26 +440,38 @@
 
     replaceContentsOfRangeInSession(uuid, resolvedRange, *fragment);
 
-    auto sessionRangeAfterReplace = contextRangeForSessionWithUUID(uuid);
-    if (!sessionRange) {
-        ASSERT_NOT_REACHED();
-        return;
-    }
     auto characterRangeAfterReplace = WebCore::CharacterRange(range.location, range.length + characterCountDelta);
 
     bool attachedToExistingSession = false;
     for (auto& session : m_textIndicatorCharacterRangesForSessions) {
         if (session.first == uuid) {
             // session.second.last() will always be valid because when creating
             // the structure we always create the vector with an element.
-            session.second.append({ replaceTextIndicatorUUID, subCharacterRange(range, session.second.last().second) });
+            session.second.append({ sourceTextIndicatorUUID, subCharacterRange(range, session.second.last().second) });
+            session.second.append({ finalTextIndicatorUUID, subCharacterRange(range, session.second.last().second) });
 
             attachedToExistingSession = true;
             break;
         }
     }
     if (!attachedToExistingSession)
-        m_textIndicatorCharacterRangesForSessions.append({ uuid, { { replaceTextIndicatorUUID, characterRangeAfterReplace } } });
+        m_textIndicatorCharacterRangesForSessions.append({ uuid, { { sourceTextIndicatorUUID, characterRangeAfterReplace }, { finalTextIndicatorUUID , characterRangeAfterReplace } } });
+
+    auto sessionRangeAfterReplace = contextRangeForSessionWithUUID(uuid);
+    if (!sessionRangeAfterReplace) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
+
+    auto replacedRangeAfterReplace = WebCore::resolveCharacterRange(*sessionRangeAfterReplace, characterRangeAfterReplace);
+
+    m_webPage->createTextIndicatorForRange(replacedRangeAfterReplace, [uuid = finalTextIndicatorUUID, weakWebPage = WeakPtr { m_webPage }](std::optional<WebCore::TextIndicatorData>&& textIndicatorData) {
```

Comment:
I think just using `finalTextIndicatorUUID` instead of assigning it to `uuid` would be better so that it doesn't shadow the other `uuid` variable